diff --git a/setup.py b/setup.py
index 297d15cb..40794da1 100644
--- a/setup.py
+++ b/setup.py
@@ -102,6 +102,8 @@ try:
 except KeyError:
     pass # no EXT_SUFFIX in this build.
 
+IS_PYPY = sys.implementation.name == "pypy"
+
 build_id_patch = build_id
 if not "." in build_id_patch:
     build_id_patch = build_id_patch + ".0"
@@ -635,6 +637,10 @@ class my_build_ext(build_ext):
         if ext.platforms and self.plat_name not in ext.platforms:
             return "Only available on platforms %s" % (ext.platforms,)
 
+        if IS_PYPY and ext.name in ('servicemanager', 'pythonservice',
+                                    'pythoncom', 'PyISAPI_loader'):
+            return "Cannot build service extensions on PyPy"
+
         # We update the .libraries list with the resolved library name.
         # This is really only so "_d" works.
         ext.libraries = patched_libs
