diff --git a/win32/src/odbc.cpp b/win32/src/odbc.cpp
index 98b828a3..f937e090 100644
--- a/win32/src/odbc.cpp
+++ b/win32/src/odbc.cpp
@@ -21,7 +21,20 @@
 #include <sql.h>
 #include <sqlext.h>
 
+#ifdef PYPY_VERSION
+PyAPI_FUNC(PyObject *) PyImport_ImportModuleLevel(
+    const char *name,           /* UTF-8 encoded string */
+    PyObject *globals,
+    PyObject *locals,
+    PyObject *fromlist,
+    int level
+    );
+
+#define PyImport_ImportModuleEx(n, g, l, f) \
+    PyImport_ImportModuleLevel(n, g, l, f, 0)
+#else
 #include <import.h>
+#endif
 
 #include <time.h>
 
